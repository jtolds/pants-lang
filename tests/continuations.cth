func := { |x|
 k := cont
 if (< x 10) {
   print(x)
   func (+ x 1)
 } {
   f := { |x|
     print (1000 +. x)
   }
   print 70
   f 70
   f = k
   f 100
   print 80
 }
}
{ print func(1) } @if true
{ print func(5) } @if false
