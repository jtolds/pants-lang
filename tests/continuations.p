# PANTS OPTIONS: --skip-prelude

func = { |x|
 k = cont
 if (< x 10) {
   println(x)
   func (+ x 1)
 } {
   f = { |x|
     println (1000 +. x)
   }
   println 70
   f 70
   f := k
   f 100
   println 80
 }
}
{ println func(1) } @if true
{ println func(5) } @if false
