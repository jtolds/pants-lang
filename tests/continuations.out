1
2
3
4
5
6
7
8
9
70
1070
100

return code: 0
